frontend/src/utilities/useCustomK8sWatchResource.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  K8sModelCommon,
+  K8sResourceCommon,
+  WatchK8sResource,
+  WebSocketOptions,
+  useK8sWatchResource,
+} from '@openshift/dynamic-plugin-sdk-utils';
+import React from 'react';
+import { CustomWatchK8sResult } from '~/types';
+
+const useCustomK8sWatchResource = <T extends K8sResourceCommon | K8sResourceCommon[]>(
+  initResource: WatchK8sResource | null,
+  initModel?: K8sModelCommon,
+  options?: Partial<WebSocketOptions & RequestInit & { wsPrefix?: string; pathPrefix?: string }>,
+): CustomWatchK8sResult<T> => {
+  const [data, loaded, error] = useK8sWatchResource<T>(initResource, initModel, options);
+
+  const loadError = React.useMemo(() => {
+    if (error instanceof Error) {
+      return error;
+    }
+
+    if (!error) {
+      return undefined;
+    }
+
+    return new Error('Unknown error occured');
+  }, [error]);
+
+  return [data, loaded, loadError];
+};
```

Comment:
Can we focus this hook on listing resources so that we can also work around the type issue in the SDK where `data` can be `undefined`.

```
const useK8sWatchResourceList = <T extends K8sResourceCommon | K8sResourceCommon[]>(
  initResource: WatchK8sResource | null,
  initModel?: K8sModelCommon,
  options?: Partial<WebSocketOptions & RequestInit & { wsPrefix?: string; pathPrefix?: string }>,
): CustomWatchK8sResult<T> => {
  const initListResource = React.useMemo(
    () => (initResource != null ? { ...initResource, isList: true } : null),
    [initResource],
  );
  const [data, loaded, error] = useK8sWatchResource<T>(initListResource, initModel, options);

  const loadError = React.useMemo(() => {
    if (error instanceof Error) {
      return error;
    }

    if (!error) {
      return undefined;
    }

    return new Error('Unknown error occured');
  }, [error]);

  // disable as data can be `undefined` by the type in the SDK is incorrect
  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
  return [React.useMemo(() => data ?? [], [data]), loaded, loadError];
};
```

If you do this, then you can also fix the issue in `templates.ts` where you have to disable the eslint rule `eslint-disable-next-line @typescript-eslint/no-unnecessary-condition`.